pkg/vdr/web/resolver.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package web
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/hyperledger/aries-framework-go/pkg/doc/did"
+	"github.com/hyperledger/aries-framework-go/pkg/framework/aries/api/vdr"
+)
+
+// Read resolves a did:web did.
+func (v *VDR) Read(didID string, opts ...vdr.ResolveOpts) (*did.Doc, error) {
+	// apply resolve opts
+	docOpts := &vdr.ResolveDIDOpts{
+		HTTPClient: &http.Client{},
+	}
+
+	for _, opt := range opts {
+		opt(docOpts)
+	}
+
+	url, err := parseDIDWeb(didID)
+	if err != nil {
+		return nil, fmt.Errorf("error resolving did:web did --> could not parse did:web did --> %w", err)
+	}
+
+	resp, err := docOpts.HTTPClient.Get(url)
+	if err != nil {
+		return nil, fmt.Errorf("error resolving did:web did --> http request unsuccessful --> %w", err)
+	}
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("error resolving did:web did --> error reading http response body: %s --> %w", body, err)
+	}
+
+	defer func() {
+		if resp.Body.Close() != nil {
+			panic("couldn't close http response body")
```

Comment:
you should not panic - just log warning with err message from body.Close()

Reply:
should I use the logger we have built in to aries instead of a plain fmt.Println for example?

Reply:
Check this out: https://github.com/hyperledger/aries-framework-go/blob/d7623b88ae7e87e4a63f5a0e3662f0d2314a0528/pkg/vdr/httpbinding/vdr.go#L108